clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -22,16 +22,32 @@ object UncommittedGarbageCollector {
   lazy val spark: SparkSession =
     SparkSession.builder().appName("UncommittedGarbageCollector").getOrCreate()
 
+  /** list repository objects directly from object store.
+   *  Reads the objects from both old repository structure and new repository structure
+   *  @param storageNamespace The storageNamespace to read from
+   *  @param before Exclude objects which last_modified date is newer than before Date
+   *  @return DF listing all objects under given storageNamespace
+   */
   def listObjects(storageNamespace: String, before: Date): DataFrame = {
+    // TODO(niro): parallelize reads from root and data paths
     val sc = spark.sparkContext
-    val dataLocation = storageNamespace + "data" // TODO(niro): handle better
+    val dataPrefix = "data"
+    val dataLocation = storageNamespace + dataPrefix // TODO(niro): handle better
```

Comment:
Is the `TODO` still relevant?

Reply:
I want to keep it - maybe we need to pass the information from the server